src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const copyArray = [...this];
+
+    for (let i = 0; i < copyArray.length; i++) {
+      if (copyArray[i] === null || copyArray[i] === undefined) {
+        copyArray[i] = '';
+      }
+
+      if (i === copyArray.length - 1) {
+        result = result + copyArray[i];
+      } else {
+        result = result + copyArray[i] + separator;
+      }
```

Comment:
and with this approach here you can just add separator here instead of those long checks